dll/cpl/sysdm/environment.c

```diff
@@ -144,16 +144,13 @@ GatherDataFromListView(HWND hwndListView,
         dwValueLength += _tcslen(szData);
     }
 
-    /* Reallocate the value buffer, regrowing it if necessary */
-    if (!VarData->lpRawValue || (_tcslen(VarData->lpRawValue) < dwValueLength))
-    {
-        if (VarData->lpRawValue)
-            GlobalFree(VarData->lpRawValue);
+    /* Reallocate the value buffer */
+    if (VarData->lpRawValue)
+        GlobalFree(VarData->lpRawValue);
 
-        VarData->lpRawValue = GlobalAlloc(GPTR, (dwValueLength + 1) * sizeof(TCHAR));
```

Comment:
(Note that we could have used the `GlobalReAlloc` function there instead for the reallocation)

Reply:
Ahh this crossed my mind entirely!